lib/api/model/model.dart

```diff
@@ -382,6 +387,19 @@ class ZulipStream {
     _$ZulipStreamFromJson(json);
 
   Map<String, dynamic> toJson() => _$ZulipStreamToJson(this);
+
+  bool hasPostingPermission(User user, {required DateTime byDate, required int realmWaitingPeriodThreshold}) {
+    final role = user.role;
+    return switch (channelPostPolicy) {
+      ChannelPostPolicy.any            => true,
+      ChannelPostPolicy.fullMembers    => role.isAtLeast(UserRole.member) && (role == UserRole.member
```

Comment:
By turning this into `||` we can replace the `... ? ... : true` clause with `... && ...`.

Reply:
I don't think I have understood it correctly. Do you mean the following?
```dart
ChannelPostPolicy.fullMembers    => role.isAtLeast(UserRole.member) || (role == UserRole.member
                                      && user.hasPassedWaitingPeriod(byDate, realmWaitingPeriodThreshold)),
```
I don't think it results in the same value. If this is not what you meant, could you please write the piece of code you suggest?

Reply:
Hmm I think you are right! I probably mistook the original expression as something else.